src/solvers/dgsem_tree/subcell_limiters.jl

```diff
@@ -41,62 +43,90 @@ The bounds are calculated using the low-order FV solution. The positivity limite
     This is an experimental feature and may change in future releases.
 """
 struct SubcellLimiterIDP{RealT <: Real, Cache} <: AbstractSubcellLimiter
+    local_minmax::Bool
+    local_minmax_variables_cons::Vector{Int}                   # Local mininum/maximum principles for conservative variables
     positivity::Bool
     positivity_variables_cons::Vector{Int}                     # Positivity for conservative variables
     positivity_correction_factor::RealT
     cache::Cache
 end
 
-# this method is used when the indicator is constructed as for shock-capturing volume integrals
+# this method is used when the limiter is constructed as for shock-capturing volume integrals
 function SubcellLimiterIDP(equations::AbstractEquations, basis;
+                           local_minmax_variables_cons = [],
                            positivity_variables_cons = [],
                            positivity_correction_factor = 0.1)
+    local_minmax = (length(local_minmax_variables_cons) > 0)
     positivity = (length(positivity_variables_cons) > 0)
 
-    bound_keys = Tuple(Symbol(string(v), "_min") for v in positivity_variables_cons)
+    bound_keys = ()
+    if local_minmax
+        for v in local_minmax_variables_cons
+            v_string = string(v)
+            bound_keys = (bound_keys..., Symbol(v_string, "_min"),
+                          Symbol(v_string, "_max"))
+        end
+    end
+    for v in positivity_variables_cons
+        if !(v in local_minmax_variables_cons)
```

Comment:
It seems like there is a hierarchy between positivity and local min/max limiting, in the sense that min/max provides a stronger guarantee than positivity. It further seems like if a variable is in local min/max, it will be ignored for positivity.

Instead of checking a doubled variable bound everywhere, why not just do it once in the constructor of the limiter and then throw an error if a variable is present in more than one limiter type? That would remove a lot of code elsewhere and make it also harder to use it in the wrong way.

Reply:
No, there is no real hierarchy here. I just wanted to make sure, that I use the same memory for equal bounds and don't have the same entry twice.

Actually, since I just use `bound_keys` to define the Dictionary anyway in 
```
for key in bound_keys
        _variable_bounds[key] = fill(nan_uEltype, n_nodes * n_nodes * capacity)
        variable_bounds[key] = unsafe_wrap(Array, pointer(_variable_bounds[key]),
                                           (n_nodes, n_nodes, capacity))
    end
```
I think doubled entries are no problem, right?

Reply:
I don't fully understand the code snippet, but I'll just say this: If you make sure in the constructor that variables are not added redundantly, you can maybe avoid checking for it in multiple other places

Reply:
Here, I just use the symbols from `bound_keys` to create memory in `variable_bounds`, where I will save the bounds later on.
It seems like in a Dictionary we cannot have doubled keys anyway. So, it should not make a difference if I add all limited variables for all limiters to `bound_keys`.

I don't really check this multiple times, do I? (Except maybe in the bounds checking routine)

Reply:
I thought I saw it in the output routines (`show`...) as well, but I might be mistaken

Reply:
No, I think the only place I do this otherwise, is in the bounds check. There it's used since the local limiting check also checks the positivity bound. So, I don't test it a second time.

Reply:
Unless you set the lower bound of the min/max limiter to a negative value, that is 😬. j/k, feel free to resolve if there is no cleaner way than what you already have

Reply:
No, there are more option the positivity bound is more restrictive (I know that's sounds stupid:sweat_smile:) But for the positivity limiting we have the parameter `positivity_correction_factor` with `u^new >= positivity_correction_factor * u^FV`.

I just changed and will commit tomorrow that I check which bound is more restrictive and only do that limiting.